flow/connectors/mysql/mysql.go

```diff
@@ -47,15 +47,15 @@ func NewMySqlConnector(ctx context.Context, config *protos.MySqlConfig) (*MySqlC
 	}, nil
 }
 
-func (c *MySqlConnector) Flavor() string {
-	switch c.config.Flavor {
-	case protos.MySqlFlavor_MYSQL_MYSQL:
-		return mysql.MySQLFlavor
-	case protos.MySqlFlavor_MYSQL_MARIA:
-		return mysql.MariaDBFlavor
-	default:
+func (c *MySqlConnector) Flavor(ctx context.Context) string {
+	// it looks like gtid_mode was never supported in MariaDB, so hoping this clears the flavor up
+	if _, err := c.Execute(ctx, "select @@gtid_mode"); err != nil {
+		if mysql.ErrorCode(err.Error()) == mysql.ER_UNKNOWN_SYSTEM_VARIABLE {
```

Comment:
better off using `errors.As` to get `mysql.MyError` so you can pull Code field instead of using this function which parses error message